Tasks/TwineAuthenticateV1/twineauthenticatemain.ts

```diff
@@ -47,91 +52,136 @@ async function main(): Promise<void> {
+                return;
+            }
+
+            (!fileContent.hasOwnProperty("distutils") || !fileContent["distutils"].hasOwnProperty("index-servers")) ?
+                fileContent["distutils"] = {["index-servers"]: feedName} : 
+                fileContent["distutils"]["index-servers"] += " " + feedName;
+
+            let encodedStr = ini.encode(fileContent);
+            fs.writeFileSync(pypircPath, encodedStr + os.EOL, 'utf8');
+
+            const feedTenant = await getFeedTenantId(feedUrl);
+            const token = await getFederatedWorkloadIdentityCredentials(entraWifServiceConnectionName, feedTenant);
+
+            if (token) {
+                tl.debug(tl.loc("Info_AddingAuthForRegistry", feedName));
+                const wifRepo = new Repository(feedName, feedUrl, 'WIFbuild', token);
+                fs.appendFileSync(pypircPath, os.EOL + wifRepo.toString(), 'utf8');
+                federatedFeedSuccessCount++;
+            }
+            else {
+                throw new Error(tl.loc("Error_FailedToGetServiceConnectionAuth", entraWifServiceConnectionName));
+            }
+
+            return;
+        }
+        else if (entraWifServiceConnectionName || feedUrl) {
+            throw new Error(tl.loc("Error_MissingFeedUrlOrServiceConnection"));
+        }
+#endif
+
+        let reposList: string[] = [];
+
+        for (let entry of newEndpointsToAdd) {
+
+            let repo = new Repository(
+                entry.packageSource.feedName,
+                entry.packageSource.feedUri,
+                entry.username,
+                entry.password
+            );
 
-                let repo = new Repository(
-                    entry.packageSource.feedName,
-                    entry.packageSource.feedUri,
-                    entry.username,
-                    entry.password
-                );
+            if ((entry.packageSource.feedName in fileContent) || (usedRepos.has(repo.repository))) {
+                console.log(tl.loc("Warning_DuplicateEntryForFeed", entry.packageSource.feedName, repo.repository));
 
-                if (usedRepos.has(repo.repository)) {
-                    tl.warning(tl.loc("DuplicateRepoUrl", repo.repository));
-                    removeFromFeedCount(internalFeed, externalEndpoints, entry);
+                if (internalFeed.has(entry)) {
+                    internalFeed.delete(entry);
                     continue;
                 }
 
-                reposList.push(repo.toString() + `${os.EOL}`);
+                externalEndpoints.delete(entry);
+                continue;
+            }
+
+            tl.debug(tl.loc("Info_AddingAuthForRegistry", entry.packageSource.feedName));
 
+            reposList.push(repo.toString() + `${os.EOL}`);
+            
+            (!fileContent.hasOwnProperty("distutils") || !fileContent["distutils"].hasOwnProperty("index-servers")) ?
+                fileContent["distutils"] = {["index-servers"]: `${entry.packageSource.feedName}`} : 
```

Comment:
is this not a duplicate of line 158?

Reply:
They are very similar, just a different feed name. I moved this into the function configHeader so there are no repeats now